dhcpv4/dhcpv4.go

```diff
@@ -230,7 +230,7 @@ func NewInform(hwaddr net.HardwareAddr, localIP net.IP, modifiers ...Modifier) (
 	)...)
 }
 
-// NewRequestFromOffer builds a DHCPv4 request from an offer.
+// NewRequestFromOffer builds a DHCPv4 request in the SELECT state from an offer.
```

Comment:
Do we need this clarification? I am not sure if this makes things more clear or more confusing.

Reply:
[RFC 2131](https://datatracker.ietf.org/doc/html/rfc2131#section-4.3.2) states that there are four different DHCPREQUEST types, this function always constructs a request in the SELECT state. This tripped me up when trying to do RENEWs with it, thus I've added the comment.

Reply:
I can however change the wording to be a bit more subtle.